src/Raven.Studio/typescript/components/pages/database/documents/identities/DocumentIdentitiesModal.tsx

```diff
@@ -0,0 +1,184 @@
+import { ModalProps } from "reactstrap/types/lib/Modal";
+import { Button, FormGroup, InputGroup, Label, Modal, ModalBody, ModalFooter } from "reactstrap";
+import React from "react";
+import { Icon } from "components/common/Icon";
+import { FormInput } from "components/common/Form";
+import { Control, SubmitHandler, useForm, useWatch } from "react-hook-form";
+import {
+    AddIdentitiesFormData,
+    DocumentIdentitiesPrefixTestContext,
+    documentIdentitiesSchema,
+} from "components/pages/database/documents/identities/DocumentIdentitiesValidation";
+import ButtonWithSpinner from "components/common/ButtonWithSpinner";
+import { useServices } from "hooks/useServices";
+import { useAppSelector } from "components/store";
+import { databaseSelectors } from "components/common/shell/databaseSliceSelectors";
+import { tryHandleSubmit } from "components/utils/common";
+import RichAlert from "components/common/RichAlert";
+import { useAsync } from "react-async-hook";
+import { LazyLoad } from "components/common/LazyLoad";
+import { yupResolver } from "@hookform/resolvers/yup";
+import { useEventsCollector } from "hooks/useEventsCollector";
+
+interface DocumentIdentitiesModalProps extends ModalProps {
+    toggleModal: () => void;
+    defaultValues?: AddIdentitiesFormData;
+    identities?: AddIdentitiesFormData[];
+    refetch: () => void;
+}
+
+export default function DocumentIdentitiesModal({
+    defaultValues,
+    refetch,
+    identities,
+    toggleModal,
+    ...props
+}: DocumentIdentitiesModalProps) {
+    const databaseName = useAppSelector(databaseSelectors.activeDatabaseName);
+    const { databasesService } = useServices();
+    const isEditing = !!defaultValues;
+    const eventsCollector = useEventsCollector();
+    const {
+        reset,
+        handleSubmit,
+        control,
+        formState: { isSubmitting },
+    } = useForm<AddIdentitiesFormData, DocumentIdentitiesPrefixTestContext>({
+        context: {
+            identities,
+            isEditing,
+        },
+        resolver: yupResolver(documentIdentitiesSchema),
+        defaultValues,
+    });
+
+    const formValues = useWatch({ control });
+
+    const onSubmit: SubmitHandler<AddIdentitiesFormData> = ({ prefix, value }) => {
+        return tryHandleSubmit(async () => {
+            await databasesService.seedIdentity(databaseName, prefix, value);
+            eventsCollector.reportEvent("identity", "new");
```

Comment:
It should report on 'Add new identity' click. Now it is also triggered on edit identity.

Reply:
Fact, I added a condition that calls the eventCollector only when we are not in edit mode